x/ibc/04-channel/keeper/handshake.go

```diff
@@ -0,0 +1,368 @@
+package keeper
+
+import (
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ics03types "github.com/cosmos/cosmos-sdk/x/ibc/03-connection/types"
+	"github.com/cosmos/cosmos-sdk/x/ibc/04-channel/types"
+	ics23 "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment"
+)
+
+// ChanOpenInit is called by a module to initiate a channel opening handshake with
+// a module on another chain.
+func (k Keeper) ChanOpenInit(
+	ctx sdk.Context,
+	order types.ChannelOrder,
+	connectionHops []string,
+	portID,
+	channelID string,
+	counterparty types.Counterparty,
+	version string,
+) (string, error) {
+	// TODO: abortTransactionUnless(validateChannelIdentifier(portIdentifier, channelIdentifier))
```

Comment:
What remains here?

Reply:
SDK-side identifier validation (up to the implementation like connection identifier validation)